libs/designsystem/src/lib/components/chart/chart.component.ts

```diff
@@ -44,152 +57,185 @@ export class ChartComponent implements OnInit, OnChanges {
     @Inject(AREASPLINE_OPTIONS) public areasplineOptions: Options,
     @Inject(TIMESERIES_OPTIONS) public timeSeriesOptions: Options,
     @Inject(ACTIVITYGAUGE_OPTIONS) public activitygaugeOptions: Options
-  ) {}
-
-  ngOnInit() {
-    this.setupChartType();
-    this.updateProperties();
-    this.chartHelper.init(this.options, this.hostElement);
+  ) {
+    this.chartHelper.init(this.hostElement);
   }
 
-  ngOnChanges() {
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['type']) {
```

Comment:
We use this syntax everywhere else:
```suggestion
    if (changes.type) {
```